### PR TITLE
Adds Logger module to the application

### PR DESCRIPTION
**Pull request by suryapratapmehra**

### CODE REVIEW

File: test/ts/station_test.exs [hunk 1]
```diff
@@ -844,7 +844,7 @@ defmodule StationTest do
 		# Send 10_000 queries
 		send_message(query, 10_000, pid)
 
-		:timer.sleep(1000)
+		:timer.sleep(5000)
```
Comment (prasadtalasila) [inline test/ts/station_test.exs]: @suryapratapmehra bumping up this number to 10_000 should make the test pass on travis.

File: lib/ts/util/itinerary.ex [hunk 2]
```diff
@@ -53,18 +53,21 @@ defmodule Util.Itinerary do
 	def is_terminal(itinerary) do
 		query = get_query(itinerary)
 		last_link = get_last_link(itinerary)
+		Logger.debug("qid :#{query.qid}; query.dst_station: #{query.dst_station}; last_link.dst_station: #{last_link.dst_station}")
```
Comment (prasadtalasila) [inline lib/ts/util/itinerary.ex]: @SuhasHebbar please edit / update the log messages put in the Itinerary module.

**Comment by prasadtalasila** [comment]
@suryapratapmehra Due to merging of PR #92, this PR got merge conflicts. Please rebase your branch.

**Comment by prasadtalasila** [comment]
@SuhasHebbar update / edit the log messages in this PR.

**Comment by SuhasHebbar** [comment]
In the debug message I think it would be good also to mention which function the debug message is originating.
```elixir
def is_terminal(itinerary) do
    query = get_query(itinerary)
    last_link = get_last_link(itinerary)

    Logger.debug(
     "In Itinerary.is_terminal\n"<>
    qid :#{query.qid}; query.dst_station: #{query.dst_station}; last_link.dst_station: #{
        last_link.dst_station
      }"
    )

    query.dst_station == last_link.dst_station
  end
```

**Comment by prasadtalasila** [comment]
@suryapratapmehra I agree. We need to find a suggested format for including the function names in the log messages. Do look at Phoenix project to see sample log messages so that we can follow it's standard format.

This PR still has a merge conflict. Please resolve it by rebasing your branch against the latest ts-v1.0-alpha branch.

File: lib/ts/util/itinerary.ex [hunk 3]
```diff
@@ -199,6 +202,7 @@ defmodule Util.Itinerary do
   # Adds a link to the present itinerary route.
   def add_link({query, route, preference}, link) do
     new_route = [link | route]
+    Logger.debug(fn -> "module=Util.Itinerary; message=Log message" end)
```
Comment (prasadtalasila) [inline lib/ts/util/itinerary.ex]: log message =  the `get_query_id(itinerary)` got `link` added to it.

File: lib/ts/util/itinerary.ex [hunk 4]
```diff
@@ -223,6 +227,7 @@ defmodule Util.Itinerary do
   # Signals iteration to stop when schedule is empty or
   # all neighbours have been visited
   defp stop_fn(neighbours, schedule) do
+    Logger.debug(fn -> "module=Util.Itinerary; message=Log message" end)
```
Comment (prasadtalasila) [inline lib/ts/util/itinerary.ex]: Log message = all neighbours visited.

Comment (prasadtalasila) [reply lib/ts/util/itinerary.ex]: A few other modifications:
* debug statement for value of should_stop on line-81
* debug statement for return value from _find_valid_connection(vars) on between lines-83 and 84

**Comment by prasadtalasila** [comment]
@suryapratap the travis wcript has to be modified to exclude the slow tests.

Add `mix test --exclude slow` to travis-script

**Pull request merged**